### PR TITLE
do not apply locality load balancer settings for inbound clusters

### PR DESCRIPTION
**Pull request by ramaraochavali**
Do not apply locality load balancer settings for inbound clusters. Fixes https://github.com/istio/istio/issues/27293
[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

File: releasenotes/notes/27293.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: traffic-management
+issue:
+  - 27293
+releaseNotes:
+  - |
+    **Fixed** Do not apply locality load balancer settings for inbound clusters.
```
Comment (linsun) [inline releasenotes/notes/27293.yaml:8]:
```suggestion
    **Fixed** a bug that locality load balancer settings were applied inbound clusters unnecessarily.
```

File: pilot/pkg/networking/core/v1alpha3/cluster.go [hunk 2]
```diff
@@ -573,10 +573,13 @@ func setH2Options(cluster *cluster.Cluster) {
 
 func applyTrafficPolicy(opts buildClusterOpts) {
 	connectionPool, outlierDetection, loadBalancer, tls := selectTrafficPolicyComponents(opts.policy)
-	applyH2Upgrade(opts, connectionPool)
+	// Connection pool settings are applicable for both inbound and outbound clusters.
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/cluster.go:576]: I thought we only call connectionPool already? https://github.com/istio/istio/blob/485a727f70b92e5cdacb62d8035570099b7f1c49/pilot/pkg/networking/core/v1alpha3/cluster_builder.go#L262

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/cluster.go:576]: https://github.com/istio/istio/blob/master/pilot/pkg/networking/core/v1alpha3/cluster_builder.go#L232 - Calls applyTrafficPolicy which is called for inbound clusters

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
Does this need cherrypick?

**Comment by ramaraochavali** [comment]
Yes. I think so. I will cherrypick to 1.6 and 1.7

**Pull request merged**